src/model.js

```diff
@@ -1283,16 +1283,21 @@ ${associationOwner._getAssociationDebugList()}`);
           const references = currentAttribute.references;
           if (currentAttribute.references) {
             const database = this.sequelize.config.database;
-            const schema = this.sequelize.config.schema;
+            const schema = tableName.schema;
+            const foreignReferenceSchema = currentAttribute.references.model.schema;
+            const foreignReferenceTableName = typeof references.model === 'object'
+              ? references.model.tableName : references.model;
             // Find existed foreign keys
             for (const foreignKeyReference of foreignKeyReferences) {
               const constraintName = foreignKeyReference.constraintName;
               if ((Boolean(constraintName)
                 && foreignKeyReference.tableCatalog === database
                 && (schema ? foreignKeyReference.tableSchema === schema : true)
-                && foreignKeyReference.referencedTableName === references.model
+                && foreignKeyReference.referencedTableName === foreignReferenceTableName
                 && foreignKeyReference.referencedColumnName === references.key
-                && (schema ? foreignKeyReference.referencedTableSchema === schema : true)
+                && (foreignReferenceSchema
```

Comment:
I think this will fail if the model has two foreign keys: one without foreignReferenceSchema, and one with it.

The one without would match either as this condition is skipped